Neos.ContentRepository/Classes/Domain/Repository/NodeDataRepository.php

```diff
@@ -1710,24 +1710,29 @@ protected function buildQueryBuilderForRelationMap($relationMap)
         $constraints = [];
         $parameters = [];
 
+        // Use a counter to avoid parameter name conflicts and limit query metadata cache entries to the maximum number of relations
+        $identifierIndex = 0;
+
         foreach ($relationMap as $relatedObjectType => $relatedIdentifiers) {
             foreach ($relatedIdentifiers as $relatedIdentifier) {
```

Comment:
you could probably just use the array key from here? but I guess fine for now.

Reply:
Don't think so, as we have two loops.